examples/nffPktgen/testing/minPktLoss.go

```diff
@@ -0,0 +1,214 @@
+        if err != nil {
+                panic(fmt.Sprintf("%s config reading failed: %v", genConfig, err))
+        }
+        context, err := generator.GetContext(configuration)
+        flow.CheckFatal(err)
+        outFlow, genChan, _ := flow.SetFastGenerator(generator.Generate, speed, context)
+        flow.CheckFatal(flow.SetSender(outFlow, uint16(outPort)))
+        if receive {
+                hc := HandlerContext {
+                        port: &portStats,
+                }
+                inFlow, err := flow.SetReceiver(uint16(inPort))
+                flow.CheckFatal(err)
+                flow.CheckFatal(flow.SetHandlerDrop(inFlow, receiveHandler, hc))
+                flow.CheckFatal(flow.SetStopper(inFlow))
+        }
+
+        go func() {
+                flow.CheckFatal(flow.SystemStart())
+        }()
+
+        // Set up finish channels
+        interruptChannel := make(chan os.Signal, 1)
+        signal.Notify(interruptChannel, os.Interrupt)
+
+        var finishChannel, statsChannel <-chan time.Time
+        if *testTime > 0 {
+                finishChannel = time.NewTicker(time.Duration(*testTime) * time.Second).C
+        }
+        if receive && *statInterval > 0 {
+                statsChannel = time.NewTicker(time.Duration(*statInterval) * time.Second).C
+        }
+
+        started := time.Now()
+
+        // optimize speed based on packet loss:
+        maxSpeed := -1
+        var totalPktRX, totalPktTX uint64 = 0, 0
+
+        go func() {
+                gen := generator.GetGenerator()
+                low := 0
+                high := 100
+                currSpeed := 100
+
+                // data at line rate:
+                portStats.packetCount = 0
+                gen.Count = 0
+                time.Sleep(time.Duration(TRAFFIC_DELAY) * time.Second)
+                pktRX := portStats.packetCount
+                pktTX := gen.GetGeneratedNumber()
+                pktLoss := calcPktLoss(pktRX, pktTX)
+                totalPktRX += pktRX
+                totalPktTX += pktTX
+                printStats(&portStats, pktRX, pktTX, pktLoss, started, currSpeed, float64(TOP_SPEED))
+
+                // binary search:
+                for low <= high {
+                        mid := (low + high) / 2
+                        currSpeed = mid
+                        updatedSpeed := 0.01 * float64(uint64(currSpeed) * TOP_SPEED)
+
+                        // reset counters:
+                        portStats.packetCount = 0
+                        gen.Count = 0
+
+                        genChan <- uint64(updatedSpeed)
+                        time.Sleep(time.Duration(TRAFFIC_DELAY) * time.Second)
+
+                        pktRX = portStats.packetCount
+                        pktTX = gen.GetGeneratedNumber()
+                        pktLoss = calcPktLoss(pktRX, pktTX)
+
+                        tgt := pktLoss <= TGT_LOSS
+                        if(tgt) {
```

Comment:
You may want to move this condition inside of `if` statement. BTW in Go you don't need to use brackets in `if`.